modules/reactor/src/meshgenerators/PolygonMeshGeneratorBase.C

```diff
@@ -0,0 +1,1229 @@
+         peripheral_index++)
+    {
+      // Loop for beginning, middle and ending positions of a side
+      for (unsigned int vector_index = 0; vector_index < 3; vector_index++)
+      {
+        sub_positions_inner.push_back(positions_inner[inner_index[peripheral_index][vector_index]]);
+        sub_d_positions_outer.push_back(
+            d_positions_outer[outer_index[peripheral_index][vector_index]]);
+      }
+      auto mesh_p = buildReplicatedMesh(2); // create an empty 2D mesh
+      auto meshp0 = build_simple_peripheral(dynamic_pointer_cast<ReplicatedMesh>(mesh_p),
+                                            num_sectors_per_side_array[pattern],
+                                            peripheral_duct_intervals[i],
+                                            sub_positions_inner,
+                                            sub_d_positions_outer,
+                                            &nodes_p,
+                                            i);
+      ReplicatedMesh other_mesh(*meshp0);
+      // rotate the peripheral mesh to the desired side of the hexagon.
+      MeshTools::Modification::rotate(other_mesh, rotation_angle, 0, 0);
+      out_meshes->prepare_for_use();
+      other_mesh.prepare_for_use();
+      out_meshes->stitch_meshes(other_mesh, OUTER_SIDESET_ID, OUTER_SIDESET_ID, TOLERANCE, true);
+      other_mesh.clear();
+      sub_positions_inner.resize(0);
+      sub_d_positions_outer.resize(0);
+    }
+  }
+  return out_meshes;
+}
+
+// 'position_setup' sets up poisitions of peripheral region before deformation due to cutoff.
+// Nine sets of positions are generated here, as shown below.
+// CORNER MESH Peripherial {0 1 2}, {2 3 4} and {4 5 6}
+//           3       2   1   0
+//            \      :   :   :
+//             \     :   :   :
+//      4.       .       :   :
+//         ` .               :
+//      5.   |               |
+//         ` |               |
+//      6.   |               |
+//         ` |               |
+//               .       .
+//                   .
+//
+// EDGE MESH Peripherial {0 1 2} and {2 7 8}
+//           8   7   2   1   0
+//           :   :   :   :   :
+//           :   :   :   :   :
+//           :   :       :   :
+//           :               :
+//           |               |
+//           |               |
+//           |               |
+//           |               |
+//               .       .
+//                   .
+void
+PolygonMeshGeneratorBase::position_setup(std::vector<std::pair<Real, Real>> * positions_inner,
+                                         std::vector<std::pair<Real, Real>> * d_positions_outer,
+                                         const Real extra_dist_in,
+                                         const Real extra_dist_out,
+                                         const Real pitch,
+                                         const unsigned int radial_index)
+{
+  (*positions_inner).resize(0);
+  (*d_positions_outer).resize(0);
+
+  (*positions_inner)
```

Comment:
is this 1) ?
This one is supposed to be right on the outer pattern boundary right?
And the outer_one is on the outer boundary of the whole thing right

Reply:
This is 0.
the inner one is the position on the unit hexagon boundary;
the outer one is the incremental position (real position = inner + outer) on the boundary of the whole thing.

The boundary of the whole thing sometimes is bigger than what is defined in the input (by `extra_dist`) so that the shape can shrink using `cut_off_hex_deform()` later to avert cutting off ring regions later.